src/components/ValueSubscriber/index.jsx

```diff
@@ -51,15 +79,27 @@ export default class ValueSubscriber extends Subscriber {
   }
 
   onChange(e) {
-    let name = this.getName();
+    let name = [];
     let value = e;
     if (e && e.target) {
       if (typeof e.target.name !== 'undefined' && e.target.name !== '') {
         name = name.concat(e.target.name);
       }
       value = e.target.value; // eslint-disable-line prefer-destructuring
     }
-    this.context.nfOnChange(fakeChangeEvent(name, value));
+
+    // Convert to a number if the current value is a number. Inputs only use
+    // string values by default, and this usually introduces some confusion.
+    if (typeof objectPath.get(this.getValue(), name) === 'number') {
+      const parsed = parseInt(value, 10);
+      // Loose equality check to make sure the value still matches
+      if (parsed == value) { value = parsed; } // eslint-disable-line eqeqeq
+    }
+
+    // Check to make sure we aren't changing the value's type
+    invariantTypesMatch(this.getName().concat(name), objectPath.get(this.getValue(), name), value);
+
+    this.context.nfOnChange(fakeChangeEvent(this.getName().concat(name), value));
```

Comment:
`nfOnChange` -> `ffOnChange` or `freeformOnChange` ?

Reply:
I created #18 and will work on it there